File: prototype-002/src/nodejs/src/components/IAwaneComponent.ts
export interface IAwaneComponent {
    readonly awaneId: string;
    
    readonly awaneLocation: string;
    
    readonly awaneInterfaces: string[];
    
    readonly awaneName: string;
    
    readonly awaneVersion: string;
    
    readonly awaneTags: Record<string, string>;
    
    awaneAs<T>(): T;
}

File: prototype-002/src/csharp/Awane.Core/Components/IAwaneComponent.cs
using System.Collections.Generic;

namespace Awane.Core.Components
{
    public interface IAwaneComponent
    {
        string AwaneId { get; }
        
        string AwaneLocation { get; }
        
        string[] AwaneInterfaces { get; }
        
        string AwaneName { get; }
        
        string AwaneVersion { get; }
        
        Dictionary<string, string> AwaneTags { get; }
        
        T? AwaneAs<T>() where T : class;
    }
}